### ISSUE TITLE
predict_uncertainty does not take transformers as an argument

### ISSUE THREAD

**Issue by amandajean119** [deepchem/deepchem#1377]
**Description:**
predict_uncertainty does not take transformers as an argument. Right now we are using undo_transforms() to generate predictions and scores in correct space, but this seems hacky.

- DeepChem Version: 2.1.0
- Platform: Mac

When predicting a regression problem we normally normalize the y vector. However, when using the predict_uncertainty function, we cannot pass the transformers data structure in, so we get transformed predictions and standard deviations. We then have to use undo_transforms to get transformed scores, but that means the standard deviations are in different units than our predicted and actual values and our scores.
`
        if self.params.uncertainty and self.params.prediction_type == 'regression' and self.params.model_type == 'NN':
            self.train_pred, self.train_std = self.model.predict_uncertainty(self.train_dataset)
            self.valid_pred, self.valid_std = self.model.predict_uncertainty(self.valid_dataset)
            self.test_pred, self.test_std = self.model.predict_uncertainty(self.test_dataset)
            self.test_std = self.test_std.flatten().tolist()
            self.train_std = self.train_std.flatten().tolist()
            self.valid_std = self.valid_std.flatten().tolist()
            if self.params.transformers:
                self.train_pred = dc.trans.undo_transforms(self.train_pred, self.transformers)
                self.valid_pred = dc.trans.undo_transforms(self.valid_pred, self.transformers)
                self.test_pred = dc.trans.undo_transforms(self.test_pred, self.transformers)
`

**Comment by peastman** [comment]
I don't know how to handle this in the general case.  The model is trained to estimate the uncertainty of its outputs.  But if you take the outputs and transform them, what happens to the uncertainty?  In your case it's just a linear transform, so you can just scale the uncertainties.  But for anything other than a linear transform, that wouldn't be valid.  That's why `predict_uncertainty()` doesn't take transformers.
